### DISCUSSION TITLE
Package Formatting for Pesticide Analysis

### DISCUSSION THREAD

**Discussion by kyle-messier** [NIEHS/PrestoGP#13] [Announcements]
@sciome-bot  @ericbair-sciome @brian-bk22

@sigmafelix  and @Spatiotemporal-Exposures-and-Toxicology  have a plan for structuring the PrestoGP package that will abide by the CRAN rules and provides reproducibility of the planned full US pesticide analysis

1.  R/ will contain R files only and be specific to the PrestoGP methodology - primarily developed by Sciome
2. vignette/ will contain a re-worked vignette from the full analysis. We are thinking we can subset to 1 state such as North Carolina. This will be a more classic vignette in that a user can walk through the analysis in a reasonable amount of time 
3. data/ will contain RDS files and other data needed to support the vignette
4. Pesticide_Analysis/ will be a folder that will live on the PrestoGP GitHub, BUT will be included in the .Rbuildignore file. This means that it is not included in the build for tarballs or CRAN. @sigmafelix will move all of our code from PrestoGP_Pesticides to this folder. This has a subfolder structure that follows 00_Load_Packages, 01_Create_Pesticdes, ..., etc. where the leading numbers provide an intuitive demarkation of workflow. For complete reproducibility, we can have the large files such as the rasters and vector for calculating covariates live on an NIEHS hosted server that can accessed by anyone willing to try to run the full analysis from the GitHub repository.  

We should be able to develop this approach and get all of the CI/CD tests passing. Then further PrestoGP/R code can continue be merged into the main branch. 

Please let us know if this sounds like a good plan.

**Comment by ericbair-sciome** [comment]
This sounds good. I just initiated a pull request with some updated testing files. This is still a work in progress, but I wanted to at least get the ball rolling. I ran into some issues testing the likelihood functions. I'll distribute an updated version once I get everything working.

This brings up a related question. It looks like there is an issue tracker for PrestoGP and another tracker/Kanban board for this project. Should I log all PrestoGP-related issues to the PrestoGP tracker? Is the board for this project used only to track the project for the pesticide data? Assuming that is correct, I am will start moving all the tracked issues on the old Teams page to the GitHub issue tracker.

**Reply by kyle-messier** [reply]
@ericbair-sciome I do not see the pull request on the PrestoGP repo. I am about to have a conversation with @sciome-bot to discuss a few things. Unless we change things, then yes, the project board that is associated with PrestoGP, called PrestoGP_Pesticides is the ideal place to start using for PrestoGP related tasks - even if it is not directly related to the pesticide analysis.

**Comment by ericbair-sciome** [comment]
@sciome-bot has to approve pull requests from the Sciome server to your GitHub. I just initiated the request an hour ago, so Shail probably hasn't had a chance to approve it and move it to your server yet. But it should happen very soon. You can nag Shail about it tonight if it hasn't happened by then. :)

**Comment by ericbair-sciome** [comment]
My new version is now updated in the sciome-main branch. By the way, am I supposed to creating a pull request between main-sciome and main when I release an updated version? Or are you and @sciome-bot handling that? I noticed that the main branch is still on version 0.2.0.9015, which is two versions behind main-sciome.
